rpcs3/rpcs3qt/settings_dialog.cpp

```diff
@@ -1057,12 +1053,40 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> gui_settings, std
 	}
 
 	ui->searchBox->setPlaceholderText(tr("Search libraries", "Library search box"));
+	ui->searchBox->setEnabled(true);
+	ui->lleList->setEnabled(true);
 
 	auto on_lib_button_clicked = [this](int id)
 	{
-		const bool enableLibs = id != static_cast<int>(lib_loading_type::liblv2only);
-		ui->searchBox->setEnabled(enableLibs);
-		ui->lleList->setEnabled(enableLibs);
+		std::vector<QListWidgetItem*> items;
+
+		// duplicate current items, we need clones to preserve checkstates
+		for (int i = 0; i < ui->lleList->count(); i++)
+		{
+			items.push_back(ui->lleList->item(i)->clone());
+		}
+
+		for (uint i = 0; i < items.size(); i++)
+		{
+			const auto text = sstr(items[i]->text());
+
+			// Select default LLE sprx list for manual lib loading, otherwise all libs are unchecked
+			items[i]->setCheckState(id == static_cast<int>(lib_loading_type::manual) && g_prx_list.count(text) && !g_prx_list.at(text)? Qt::Checked : Qt::Unchecked);
+		}
+
+		// sort items: checked items first then alphabetical order
+		std::sort(items.begin(), items.end(), [](QListWidgetItem* i1, QListWidgetItem* i2)
+		{
+			return (i1->checkState() != i2->checkState()) ? (i1->checkState() > i2->checkState()) : (i1->text() < i2->text());
+		});
+
+		// refill library list
+		ui->lleList->clear();
+
+		for (uint i = 0; i < items.size(); i++)
+		{
+			ui->lleList->addItem(items[i]);
```

Comment:
There is an undefined behavior:
The search filter is not considered when changing the lib mode...

Reply:
Fixed, fixed, fixed, fixed, fixed and fixed.